decaf377-fmd/src/clue_key.rs

```diff
@@ -99,4 +109,10 @@ impl ExpandedClueKey {
 
         Ok(Clue(buf))
     }
+
+    /// Checks that the expanded clue key has at least `precision` subkeys
+    fn ensure_at_least(&self, precision: usize) -> bool {
+        // TODO: handle try_borrow failure via Result
+        return self.subkeys.try_borrow().unwrap().len() >= precision;
+    }
```

Comment:
This `ensure_at_least` just checks whether the existing expansion was large enough to handle the specified precision, but it would be more convenient as an API user to not have to think about the expansion at all, and have it happen automatically.

To do that, I think the expansion logic from the current `expand` function should move into `ensure_at_least`, so that `ensure_at_least` will expand the key up to the specified precision.  This means that the `ExpandedClueKey` would need to store the `root_pub: decaf377::Element`, so that it's accessible to `ensure_at_least`.

The existing `expand` function does three things:

1. It converts a `ClueKey` to an `ExpandedClueKey`;
2. It validates that the bytes of the `ClueKey` are a valid decaf377 encoding, and decodes them to a group element;
3. It does the expansion.

With the new `ensure_at_least` behavior, the user wouldn't have to think about expansion at all, so the only remaining behaviors are (1) and (2).  In that case, I think it would make sense to rename it to `ExpandedClueKey::new(clue_key: &ClueKey) -> Result<Self, ...>`, and to provide a convenience wrapper `ClueKey::expand(&self) -> Result<ExpandedClueKey, ...>` that calls `ExpandedClueKey::new(&self)`.

Reply:
Got it, thanks for your feedback; this is helpful and an improvement for sure!